### ISSUE TITLE
clean up tt_memory span packing

### ISSUE THREAD

**Issue by pgkeller** [tenstorrent/tt-metal#13944]
Presently spans are packed by looking at the firmware base addresses from dev_mem_map.h to determine if a span is in the text or data section.  The new elf loader natively understands text/data sections so we can use this info to avoid this ugly inference pattern.

**Comment by blozano-tt** [comment]
@pgkeller @nathan-TT @jbaumanTT is this still true?

I don't notice any usage of dev_mem_map.h in tt_memory.cpp or tt_memory.hpp

I see an include of tensix.h but I think that is dead code.

Maybe we can close this?

**Comment by nathan-TT** [comment]
I have not started investigating

**Comment by pgkeller** [comment]
no, can't close.  a recent change put that code behind the hal, but it needs to be deleted in favor of a better methodology (no need to depend on the dev mem map at all)

**Comment by nathan-TT** [comment]
yes, I thnk so
